cfme/tests/containers/test_properties.py

```diff
@@ -156,27 +155,24 @@ def test_properties(provider, test_item, soft_assert):
                                    .format(test_item.obj.__name__, instance.name, field))
 
 
-@pytest.mark.skip(reason="This test is currently skipped due to instability issues. ")
-def test_pods_conditions(provider, appliance, soft_assert):
+def test_pods_conditions(provider, appliance, pods_per_ready_status, soft_assert):
 
-    #  TODO: Add later this logic to mgmtsystem
     selected_pods_cfme = {pd.name: pd
                           for pd in Pod.get_random_instances(
                               provider, count=3, appliance=appliance)}
 
-    selected_pods_ose = {pod["metadata"]["name"]: pod for pod in
-                         provider.mgmt.api.get('pod')[1]['items'] if pod["metadata"]["name"] in
-                         selected_pods_cfme}
-
     for pod_name in selected_pods_cfme:
         cfme_pod = selected_pods_cfme[pod_name]
 
-        ose_pod = selected_pods_ose[pod_name]
-
-        ose_pod_condition = {cond["type"]: cond["status"] for cond in
-                             ose_pod['status']['conditions']}
-        cfme_pod_condition = {type: getattr(getattr(cfme_pod.summary.conditions, type), "Status")
-                              for type in ose_pod_condition}
-
-        for item in cfme_pod_condition:
-            soft_assert(ose_pod_condition[item], cfme_pod_condition[item])
+        ose_pod_condition = pods_per_ready_status[pod_name]
+        cfme_pod_condition = {_type:
+                              eval_strings(
+                                  [getattr(getattr(cfme_pod.summary.conditions, _type), "Status")]
+                              ).pop()
+                              for _type in ose_pod_condition}
+
+        for status in cfme_pod_condition:
+            soft_assert(ose_pod_condition[status] == cfme_pod_condition[status],
+                        'Expected "{}" status for pod {} is {}. Found {} instead.'
```

Comment:
The English grammar is bad in the error message and it doesn't match the assert condition. Why it is not saying something like

    The Pod {} status mismatch: It is "{}" in openshift while cfme sees "{}".

Not mentioning that it doesn't make much sense to me why one needs to first group the pods by it status to compare them then. It smells like the Polarion prescription is bad, but I cannot check it in the moment.